Make the shit less shitty (#1)
mg-code-reviewer/drone-controller-challenge-java: README.md

# Drone Controller Challenge

## Description

Our merchant has a drone flying in the field and spraying fertilizer to protect their crops.
Their field is divided in sectors of 10 x 10. Our merchant can control this Drone via REST by commands:

* Turn left or right (response example I am on [2, 3] watching on North (or West))
* Go forward (response Now I am at [2, 3] Or can not do it [9, 9])
And they can ask for the drone’s status:

> * Where are you? (response [2, 3])

Reviewer: are examples in task supposed to be bad practice too? to me this is too confusing. at least the requirements should look fine

Reviewer 2: +1 for that, I would suggest at least to show what we're expecting as a response (e.g. a JSON example)

Author: I made a fix, please check it out and fix if you see any issues :pray:

## Task
We ask you to identify and fix as many issues as possible in the code we provided. 
You should also pay attention to make notes with all the recommendations for the person who developed this solution. 

### Constraints

> * The drone can look only in 4 directions (North, South, West, East)

Reviewer: imho we should remove 'only' as it is limiting too much mental model of implementation and suggest not to focus on future changes

* The drone can only turn to neighbouring directions (e.g., if the drone is facing North it
can turn only to West or East)

> ## Delivery
> Please copy the template repository. We ask for you to return this task to us no more
> than 7 days after the day that you receive it. We recommend to not spend more than 4 hours in total.
> Once you are happy with your output, please share a private GitHub repository of the code
> with the following account: **mg-code-reviewer**
> Good luck and thank you for your time and consideration in being our next SumUpper!

Reviewer 2: I find it a bit more clear - but let me think if you have other opinions 🙏 

```suggestion
## Delivery
Please create your own private repository from this template repository. We ask for
 you to return this task to us no more than 7 days after the day that you receive it.
 We recommend not spending more than 4 hours in total.
Once you are happy with your output, please share the GitHub repository of the code
with the following account: `mg-code-reviewer`
Good luck and thank you for your time and consideration in being our next SumUpper!
```

# How run the project

The project could be run by the following command:

> `./gradlew bootRun`

Reviewer 2: Nitpick, I would suggest putting this in a structured way such as:

````shell
./gradlew bootRun
````

> # Recommendations

Reviewer: we should be more specific with our expectations. if this is write _todo for general development_, then it should be it. if it's _steps to make it prod ready_ - let's put it

Author: Done
